src/analyze.cpp

```diff
@@ -3754,49 +3754,59 @@ static void analyze_fn_body(CodeGen *g, ZigFn *fn_table_entry) {
     analyze_fn_ir(g, fn_table_entry, return_type_node);
 }
 
-static void add_symbols_from_import(CodeGen *g, AstNode *src_use_node, AstNode *dst_use_node, ScopeDecls* decls_scope) {
+static void add_symbols_from_struct(CodeGen *g, AstNode *src_use_node, AstNode *dst_use_node, ScopeDecls* decls_scope) {
     if (src_use_node->data.use.resolution == TldResolutionUnresolved) {
         preview_use_decl(g, src_use_node, decls_scope);
     }
 
-    ConstExprValue *use_target_value = src_use_node->data.use.using_namespace_value;
-    if (type_is_invalid(use_target_value->type)) {
+    ConstExprValue *use_expr = src_use_node->data.use.using_namespace_value;
+    if (type_is_invalid(use_expr->type)) {
         decls_scope->any_imports_failed = true;
         return;
     }
 
     dst_use_node->data.use.resolution = TldResolutionOk;
 
-    assert(use_target_value->special != ConstValSpecialRuntime);
+    assert(use_expr->special != ConstValSpecialRuntime);
 
-    ZigType *target_import = use_target_value->data.x_type;
-    assert(target_import);
+    // The source struct for the imported symbols
+    ZigType *src_ty = use_expr->data.x_type;
+    assert(src_ty);
 
-    if (target_import->id != ZigTypeIdStruct) {
+    if (src_ty->id != ZigTypeIdStruct) {
```

Comment:
```suggestion
    if (src_ty->id != ZigTypeIdStruct || is_slice(src_ty)) {
```

Reply:
Is that worth preventing?
e.g. if a user really want `ptr` and `len` in scope, why stop them?

Reply:
> Is that worth preventing?

I guess so, this way we leak an implementation detail.

Reply:
@daurnimator `use` does not import fields, so `use []u8` does nothing. It's an implementation detail that a slice is represented as a struct internally, so I think it is correct to make this an error.

Reply:
if #1214 is implemented, could `use []T` be a way to implement #173?